### DISCUSSION TITLE
What role does the Canny board actually play?

### DISCUSSION THREAD

**Discussion by Josh-Cena** [facebook/docusaurus#5255] [General] [answered]
I know that it's used for immature feature requests, but:

- Are potential contributors supposed to go there and pick requests to work on?
- Is someone expected to polish the request and turn it into a formal RFC issue on GitHub?
- Is there a "threshold of upvotes" that, once reached, the maintainers would come in and work on it? (Or at least give some detailed feedback?)

For one thing, the maintainers haven't participated in the discussion for long (I only saw Endi occasionally), making it hard to guess whether these features are worth consideration.

**Comment by slorber** [comment, marked as answer]
I think things were a bit messy because bugs/feature requests are often both opened on GitHub and Canny.

I think Canny is great to get an idea about how much traction there is for a feature request. 
One can say Github issues also have reactions.

I'm trying to route more GitHub "ideas" to Canny recently, and will also try to keep it more up-to-date (which I haven't been very good at so far):

![image](https://user-images.githubusercontent.com/749374/127660120-0017eca4-79cb-45d8-abc7-b4948ef6def0.png)

![image](https://user-images.githubusercontent.com/749374/127660163-eacbf46c-9a90-41c0-a38d-a6b19ee10551.png)

In general, I don't have a strong opinion on what should go where exactly, just trying to organize things. Maybe Canny is not that useful and we should drop it? Or maybe canny is useful and we should move more things there? I don't really know for sure.


---

> Are potential contributors supposed to go there and pick requests to work on?

In general, I'd say anything significant should have a design phase so that we can give you at least some feedback and make sure your planned implementation does not conflict with other planned APIs and the general direction we want to take for Docusaurus

> Is someone expected to polish the request and turn it into a formal RFC issue on GitHub?

Anyone is welcome to open an RFC and attempt to define exactly what the feature should do and what is its public API. This opens the discussion, highlight edge cases and make it more likely to have an upcoming implementation.

> Is there a "threshold of upvotes" that, once reached, the maintainers would come in and work on it? (Or at least give some detailed feedback?)

Not particularly. II just gives an indication of how many people want this feature to happen but ultimately we already have an existing roadmap with probably even more important things to work on, and we have limited bandwidth to implement everything.

> For one thing, the maintainers haven't participated in the discussion for long (I only saw Endi occasionally), making it hard to guess whether these features are worth consideration.

I'll try to answer more and also keep it up to date. 
Some features are somehow already implemented by the community through plugins so it's not clear to me if we should mark them as completed or if people really want us to maintain some "official" thing in the main repo. We can't really support all use-cases in the main repo, but I'm trying to get a "docusaurus-community" repo so that we can move at least some plugins and federate better our community around some well-maintained community plugins, also making sure that I can myself fix major issues more easily if needed.

I guess all features in Canny with upvotes are worth considering, but it does not mean we can implement all of them ourselves in the main repo in the end, and we need the help of the community. Canny can also help community members to figure out if it's worth building a community plugin for a certain use-case.

---

In general, I'd like to inform more the community about the released and upcoming features. I suggested that we could open a newsletter and do quarterly blog posts for that with release highlights (generally found in tweets) and features scheduled in the near future.

Also, if you are looking for impactful things to work on, we can definitively figure this out on Discord

**Reply by Josh-Cena** [reply]
Thanks for the explanation. Just asking out of curiosity because seeing the feature request issue template emphasizing a lot about Canny yet it doesn't seem so popular...
